pkg/cli/admin/oauth/oauth.go

```diff
@@ -0,0 +1,261 @@
+	o.Duration = time.Hour
+	cmd := &cobra.Command{
+		Use:   name,
+		Short: "Create a new OAuth access token",
+		Long: templates.LongDesc(`
+			Create an OAuth access token
+
+			This command will create an OAuth access token with the requested scopes and expiration time for the
+			current user or a requested user. The actions for creating a token require cluster admin and will
+			be assigned to an OAuth client called 'explicit-token-grant' which will not be usable from normal
+			OAuth flows.
+
+			The scopes that may be assigned to a token are:
+			
+			  * user:info - retrieve user name and uid from the OAuth server
+			  * user:full - act as the user
+			  * role:<cluster_role_name>:* - actions allowed by the provided cluster role in any namespace
+			  * role:<cluster_role_name>:<namespace> - actions allowed by the provided cluster role in the provided
+
+			The token will be printed on success and may be used via 'oc login' or the '--token=' argument.
+
+			Note that this command will have no effect if the OAuth component has been disabled via configuration.
+			`),
+		Example: fmt.Sprintf(createTokenExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+	cmd.Flags().StringVar(&o.UserName, "user", o.UserName, "The name of the user to create a token for. Defaults to the current user.")
+	cmd.Flags().StringVar(&o.RoleScope, "role", o.RoleScope, "Sets the scope to the provided cluster role name in the current namespace.")
+	cmd.Flags().BoolVar(&o.RoleAll, "role-all", o.RoleAll, "Sets the role scope over all namespaces.")
+	cmd.Flags().StringSliceVar(&o.UserScopes, "scopes", o.UserScopes, "The scopes to be assigned to the user.")
+	cmd.Flags().DurationVar(&o.Duration, "duration", o.Duration, "How long before this token expires.")
+
+	kcmdutil.AddDryRunFlag(cmd)
+	o.PrintFlags.AddFlags(cmd)
+	return cmd
+}
+
+func (o *CreateTokenOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	var err error
+	o.DryRunStrategy, err = kcmdutil.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+	o.Output = kcmdutil.GetFlagString(cmd, "output")
+
+	o.ToPrinter = func(message string) (printers.ResourcePrinter, error) {
+		o.PrintFlags.NamePrintFlags.Operation = message
+		kcmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+		return o.PrintFlags.ToPrinter()
+	}
+
+	clientConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	userClient, err := userclient.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+	oauthClient, err := oauthclient.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+	o.UserInterface = userClient.UserV1().Users()
+	o.OAuthAccessTokenInterface = oauthClient.OauthV1().OAuthAccessTokens()
+	o.OAuthClientInterface = oauthClient.OauthV1().OAuthClients()
+
+	if len(o.UserUID) == 0 {
+		userName := o.UserName
+		if len(userName) == 0 {
```

Comment:
You set this to "~" above, you should add validation preventing it from being set to an empty string